api-docs/src/content.config.ts

```diff
@@ -0,0 +1,14 @@
+import { docsLoader } from '@astrojs/starlight/loaders';
+import { docsSchema } from '@astrojs/starlight/schema';
+import { defineCollection, z } from 'astro:content';
+
+export const collections = {
+  docs: defineCollection({
+    loader: docsLoader(),
+    schema: docsSchema({
+      extend: z.object({
+        pageType: z.optional(z.string()),
```

Comment:
Customizes the page schema to add `pageType`. This is used to only render the Markdown button / links on element pages, but not on the index page.